### PR TITLE
Small IrCloner fix

### PR DESCRIPTION
**Pull request by csarofeen**
Don't have IrCloner copy the is fusion input/output member of Val's because we use it now for recomputing expressions internally to a fusion, so don't want to set another tensor to an output just because we decided to recompute the output.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/fusion.cpp [hunk 1]
```diff
@@ -255,7 +261,11 @@ void Fusion::replaceOutput(Val* output, Val* replacement) {
   TORCH_CHECK(find_output != outputs_.end(), "Unable to find output in Fusion");
 
   if (find_output != outputs_.end()) {
-    *find_output = replacement;
+    std::replace_if(
+        outputs_.begin(),
+        outputs_.end(),
+        [&output](Val* v) { return v == output; },
+        replacement);
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/fusion.cpp:268]: Is this changed for when `outputs_` has multiple vals that are equal to `output`?

File: torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp [hunk 2]
```diff
@@ -90,9 +90,7 @@ Val::Val(IrBuilderPasskey passkey, ValType _vtype, DataType _dtype)
 Val::Val(const Val* src, IrCloner* ir_cloner)
     : Statement(src, ir_cloner),
       vtype_(src->vtype_),
-      dtype_(src->dtype_),
-      is_fusion_input_(src->is_fusion_input_),
-      is_fusion_output_(src->is_fusion_output_) {}
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp:95]: Wow, nice catch. How could this have been working so far :astonished:

**Review by naoyam — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp (hunk 2 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp:95]: I don't know, this was definitely on the scarier side of bugs.

File: torch/csrc/jit/codegen/cuda/fusion.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/fusion.cpp:268]: No, I just thought replace_if seemed like better syntax for the operator. I didn't really like the iterator dereference + set. I thought that could have been the issue.

Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/fusion.cpp:268]: If someone sets the same output twice... that would probably be.... not good.

**Pull request merged**